### PR TITLE
Categorize the max text line length exceeded error as HIVE_BAD_DATA

### PR DESCRIPTION
**Pull request by yingsu00**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -374,7 +374,7 @@
             <dependency>
                 <groupId>com.facebook.presto.hadoop</groupId>
                 <artifactId>hadoop-apache2</artifactId>
-                <version>2.7.4-4</version>
+                <version>2.7.4-5-SNAPSHOT</version>
```
Comment (sopel39) [inline pom.xml]: Could you make a new release of `hadoop-apache2`

Comment (yingsu00) [reply pom.xml]: @electrum David will help make a release of hadoop-apache2. Thanks David!

File: presto-hive/src/main/java/com/facebook/presto/hive/GenericHiveRecordCursor.java [hunk 2]
```diff
@@ -212,6 +214,9 @@ public boolean advanceNextPosition()
         }
         catch (IOException | SerDeException | RuntimeException e) {
             closeWithSuppression(this, e);
+            if (e instanceof TextLineLengthLimitExceededException) {
+                throw new PrestoException(HIVE_BAD_DATA, e);
```
Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/GenericHiveRecordCursor.java]: Let's add a more user friendly error message:
```java
throw new PrestoException(HIVE_BAD_DATA, "Line too long in text file: " + path, e);
```
You'll need to pass the path into the constructor.

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java [hunk 3]
```diff
@@ -216,6 +218,10 @@ private HiveUtil()
             return recordReader;
         }
         catch (IOException e) {
+            if (e instanceof TextLineLengthLimitExceededException) {
+                throw new PrestoException(HIVE_BAD_DATA, e);
```
Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java]: Same here

File: presto-hive/src/main/java/com/facebook/presto/hive/GenericHiveRecordCursor.java [hunk 4]
```diff
@@ -115,14 +119,17 @@ public GenericHiveRecordCursor(
             Properties splitSchema,
             List<HiveColumnHandle> columns,
             DateTimeZone hiveStorageTimeZone,
-            TypeManager typeManager)
+            TypeManager typeManager,
+            Path path)
```
Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/GenericHiveRecordCursor.java]: Move to first argument of constructor

**Pull request merged**